File: Pythoncode/20.有效的括号.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for x in s:
            if len(stack) == 0:
                stack.append(x)
            elif stack[-1] == "(" and x == ")":
                stack.pop()
            elif stack[-1] == "{" and x == "}":
                stack.pop()
            elif stack[-1] == "[" and x == "]":
                stack.pop()
            else:
                stack.append(x)
        return len(stack) == 0
        
s = Solution()
s.isValid("()")

File: Cppcode/20.有效的括号.cpp
#include <string>
#include <stack>

using namespace std;
class Solution {
public:
    bool isValid(string s) {
        stack<char> l;
        for (char c : s)
        {
            if (c == '(' or c == '{' or c == '[') l.push(c);
            else if (l.empty()) return false;
            else if ((c == ')' and l.top() == '(') or (c == '}' and l.top() == '{') or (c == ']' and l.top() == '[')) l.pop();
            else return false;
        }
        if (l.empty()) return true;
        else return false;
    }
};